### PR TITLE
feat(design): make qty-dropdown use Angular Forms bindings

### PR DESCRIPTION
**Pull request by damienwebdev**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

**Comment by lderrickable** [comment]
Also, can you serve `demo`? I'm getting an error.

File: apps/design-land/src/app/qty-dropdown/qty-dropdown.module.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { QtyDropdownComponent } from './qty-dropdown.component';
+import { DesignLandQtyDropdownRoutingModule } from './qty-dropdown-routing.module';
+import { DaffQtyDropdownModule } from '@daffodil/design';
+import { DaffFormFieldModule } from 'libs/design/src';
```
[the remark this answers is not in the document]
Comment (damienwebdev) [reply apps/design-land/src/app/qty-dropdown/qty-dropdown.module.ts]: fixed in 2da78f676077f3240a1ac95c381aed8139a19acb

File: libs/design/src/atoms/form/form-field/form-field/form-field.component.ts [hunk 2]
```diff
@@ -47,7 +47,7 @@ export class DaffFormFieldComponent implements DoCheck, AfterContentInit, AfterC
    * observable to remove unnecessary change detection.
    */
   ngDoCheck() {
-    if(this._control) {
+    if(this._control!.ngControl) {
```
[the remark this answers is not in the document]
Comment (damienwebdev) [reply libs/design/src/atoms/form/form-field/form-field/form-field.component.ts]: @lderrickable what I didn't see until a few days ago is that there's actually a bug here. In the event the control doesn't actually use an `ngControl` we'll get a null reference exception at this line. @xelaint has a branch to address this, but this was an initial attempt when I found the bug the first time. I'll revert this and depend on @xelaint's work.

**Comment by damienwebdev** [comment]
@lderrickable I seem to be able to serve it just fine, what's the exact error you're getting?

**Comment by damienwebdev** [comment]
Closing in favor of #394

**Pull request closed without merging**